### PR TITLE
API-6680 - Upgrading Scala to 2.13

### PR DESCRIPTION
**Pull request by peteslater-ee**

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -31,7 +31,7 @@ newlines.implicitParamListModifierPrefer = before
 newlines.avoidForSimpleOverflow = [tooLong, punct, slc]
 
 rewrite.rules = [SortModifiers]
-rewrite.rules = [SortImports]
+rewrite.rules = [AsciiSortImports]
```
Comment (AndySpaven) [inline .scalafmt.conf]: aren't we doing this the new way?  Not really fussed

File: app/uk/gov/hmrc/testuser/models/EnumJson.scala [hunk 2]
```diff
@@ -20,15 +20,15 @@ import play.api.libs.json._
 
 object EnumJson {
 
-  def enumReads[E <: Enumeration](enum: E): Reads[E#Value] = new Reads[E#Value] {
+  def enumReads[E <: Enumeration](`enum`: E): Reads[E#Value] = new Reads[E#Value] {
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/testuser/models/EnumJson.scala]: Is there a reason not to name it enumValue ?

File: test/uk/gov/hmrc/testuser/common/SuppressedLogFilter.scala [hunk 3]
```diff
@@ -27,11 +26,11 @@ import ch.qos.logback.core.spi.FilterReply
 import play.api.LoggerLike
 
 class SuppressedLogFilter(val messagesContaining: String) extends Filter[ILoggingEvent] {
-  private val suppressedEntries = new mutable.MutableList[ILoggingEvent]()
+  private var suppressedEntries = List[ILoggingEvent]()
```
Comment (AndySpaven) [inline test/uk/gov/hmrc/testuser/common/SuppressedLogFilter.scala:29]: Even I'm not that immutable :-)

File: app/uk/gov/hmrc/testuser/models/EnumJson.scala (hunk 2 above)
Comment (peteslater-ee) [reply app/uk/gov/hmrc/testuser/models/EnumJson.scala]: `enum` is 3 less characters than enumValue?

**Review by AndySpaven — APPROVED** [review]

**Comment by peteslater-ee** [comment]
rebuild

**Pull request merged**